### ISSUE TITLE
Unused import not caught depending on import method

### ISSUE THREAD

**Issue by jsundahl** [PyCQA/flake8#1789]
### how did you install flake8?

```console
pip install flake8 (on python 3.11.0)
```

### unmodified output of `flake8 --bug-report`

```json
{
  "platform": {
    "python_implementation": "CPython",
    "python_version": "3.11.0",
    "system": "Linux"
  },
  "plugins": [
    {
      "plugin": "mccabe",
      "version": "0.7.0"
    },
    {
      "plugin": "pycodestyle",
      "version": "2.10.0"
    },
    {
      "plugin": "pyflakes",
      "version": "3.0.1"
    }
  ],
  "version": "6.0.0"
}
```

#### setup

these are the files we will be linting:
```
$ cat main.py
import foo.used
import foo.unused

foo.used.something = "something"
```
```
$ cat main_caught.py
from foo import used
from foo import unused

used.something = "something"

```

create the files it imports
```
$ mkdir foo
$ touch foo/unused.py
$ touch foo/used.py
```

#### actual behavior
unused import in main.py is not caught
```
$ flake8 main.py 
main.py:5:1: W391 blank line at end of file
```

unused import in main_caught.py is caught
```
beefy:~$ flake8 main_caught.py 
main_caught.py:2:1: F401 'foo.unused' imported but unused
main_caught.py:5:1: W391 blank line at end of file
```

#### expected behavior
`foo.unused` should be marked as unused in `main.py`
